### DISCUSSION TITLE
DICOMSeriesToVolumeOperator in MR modality

### DISCUSSION THREAD

**Discussion by macayaven** [Project-MONAI/monai-deploy-app-sdk#218] [Q&A] [answered]
Hi everyone, @MMelQin @gigony @GreySeaWolf 

I'm trying to create a toy image registration sample using the app-sdk + niftireg using some MR modality images and I'm having some issues that I would like to share with you and get your thoughts. 

It seems like DICOMSeriesToVolumeOperator fails to complete due to the absense of Rescale Intercept attribute. I am used to work with CT modality (in which this DICOM tag is conditionally required), so I am not sure if this is a problem with the DICOM series (although I think the series is OK as I am able to open it in a viewer like Aliza).
Could it be that DICOMSeriesToVolumeOperator need to be tweeked in order to handle MR modality? Or do you think this is just a problem with this specific series?

I'm attaching a .zip containing a notebook with the steps to reproduce the error. [01_toy_registration_sample.zip](https://github.com/Project-MONAI/monai-deploy-app-sdk/files/7639690/01_toy_registration_sample.zip)

Thank you!

**Error output:**
When I execute:
```
op = DICOMSeriesToVolumeOperator()
image = op.convert_to_image(study_selected_series_list)
```
I get this error:
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-111-f43aa482a2c0> in <module>
      1 op = DICOMSeriesToVolumeOperator()
----> 2 image = op.convert_to_image(study_selected_series_list)

/usr/local/share/methinks/anaconda3/envs/monai/lib/python3.9/site-packages/monai/deploy/operators/dicom_series_to_volume_operator.py in convert_to_image(self, study_selected_series_list)
     59             metadata.update(selection_metadata)
     60 
---> 61             voxel_data = self.generate_voxel_data(dicom_series)
     62             image = self.create_volumetric_image(voxel_data, metadata)
     63 

/usr/local/share/methinks/anaconda3/envs/monai/lib/python3.9/site-packages/monai/deploy/operators/dicom_series_to_volume_operator.py in generate_voxel_data(self, series)
     85         vol_data = np.stack([np.transpose(s.get_pixel_array()) for s in slices], axis=-1)
     86         vol_data = vol_data.astype(np.int16)
---> 87         intercept = slices[0][0x0028, 0x1052].value
     88         slope = slices[0][0x0028, 0x1053].value
     89 

/usr/local/share/methinks/anaconda3/envs/monai/lib/python3.9/site-packages/monai/deploy/core/domain/dicom_sop_instance.py in __getitem__(self, key)
     41 
     42     def __getitem__(self, key: Union[int, slice, Tag]) -> Union[Dataset, DataElement]:
---> 43         return self._sop.__getitem__(key)
[01_toy_registration_sample.zip](https://github.com/Project-MONAI/monai-deploy-app-sdk/files/7639649/01_toy_registration_sample.zip)

     44 
     45     def get_pixel_array(self):

/usr/local/share/methinks/anaconda3/envs/monai/lib/python3.9/site-packages/pydicom/dataset.py in __getitem__(self, key)
    914                 raise KeyError(f"'{key}'") from exc
    915 
--> 916         elem = self._dict[tag]
    917         if isinstance(elem, DataElement):
    918             if elem.VR == 'SQ' and elem.value:

KeyError: (0028, 1052)
```

**Comment by MMelQin** [comment, marked as answer]
Thank you @macayaven Carlos for your feedback and question.

Yes, I did encounter the exact error, and had created a [bug ](https://github.com/Project-MONAI/monai-deploy-app-sdk/issues/187) more than a month ago, but somehow, did not get to fix it in time for the App SDK v0.2 release. Will schedule this bug for v0.3, the latest.

For sure, the DICOM series in not to blame. When the intercept and/or slope are not present, it is safe for the processor to use the defaults, namely 0 for intercept and 1 for slope respectively.

Thanks also for mentioning Aliza. Some of the working group members are using it, too. Outside of the commercial products, I typically use 3D Slicer, MicromDicom, and sometimes Orthanc (its viewer is really limited but parser is good).

The bug has been fixed in #236

**Reply by macayaven** [reply]
Great Ming, thank you for your response!
